src/sender.rs

```diff
@@ -14,35 +14,47 @@ use crate::{
 
 pub struct Sender {
     frame_count: FrameCountGenerator,
-    sender_id: KeyId,
+    key_id: KeyId,
     cipher_suite: CipherSuite,
     secret: Option<Secret>,
     buffer: Vec<u8>,
 }
 
 impl Sender {
-    pub fn new(sender_id: u64) -> Sender {
-        Self::with_cipher_suite(sender_id, CipherSuiteVariant::AesGcm256Sha512)
+    pub fn new<K>(key_id: K) -> Sender
+    where
+        K: Into<KeyId>,
```

Comment:
`I` for Id?

Reply:
Using `Id`now

Reply:
a matter of taste really. I personally like single character type-parameters, as you can more clearly distinguish them from actual types.

Reply:
Yes, a good topic for endless discussions :D